### DISCUSSION TITLE
Failed install libsystemd with x64-linux on Ubuntu 22.04 x86_64

### DISCUSSION THREAD

**Discussion by ArthurChan2018** [microsoft/vcpkg#35277] [Q&A] [answered]
I install libsystemd  libsystemd with x64-linux on Ubuntu 22.04 x86_64 failed.
I'd appreciate advice how to resolve the issue

**Host Environment**

> Host: Ubuntu 22.04 x86_64
> package: libsystemd v254
> triplet: x64-linux
> Compiler: GNU 11.4.0
> vcpkg version:  23ceb9cbf9b6d32f485cf039547b70102a6ef9d8 (2023-11-23)

**Failure logs**

> ninja: Entering directory `vcpkg/buildtrees/libsystemd/x64-linux-dbg'
> [1/191] Generating src/basic/filesystems-gperf.h with a custom command
> [2/191] Generating version.h with a custom command
> [3/191] Generating src/libsystemd/audit_type-list.txt with a custom command (wrapped by meson to capture output)
> [4/191] Generating src/basic/arphrd-list.txt with a custom command (wrapped by meson to capture output)
> [5/191] Generating src/basic/filesystem-list.h with a custom command (wrapped by meson to capture output)
> [6/191] Generating src/basic/af-list.txt with a custom command (wrapped by meson to capture output)
> [7/191] Generating src/basic/cap-list.txt with a custom command (wrapped by meson to capture output)
> [8/191] Generating src/basic/filesystem-switch-case.h with a custom command (wrapped by meson to capture output)
> [9/191] Generating src/basic/errno-list.txt with a custom command (wrapped by meson to capture output)
> [10/191] Generating src/libsystemd/libsystemd.pc with a custom command
> FAILED: src/libsystemd/libsystemd.pc 
> vcpkg/buildtrees/libsystemd/src/v254-832da4078b.clean/tools/meson-render-jinja2.py config.h version.h ../src/v254-832da4078b.clean/src/libsystemd/libsystemd.pc.in src/libsystemd/libsystemd.pc
> Traceback (most recent call last):
>   File "vcpkg/buildtrees/libsystemd/src/v254-832da4078b.clean/tools/meson-render-jinja2.py", line 9, in <module>
>     import jinja2
> ModuleNotFoundError: No module named 'jinja2'
> [11/191] Generating src/libudev/libudev.pc with a custom command
> FAILED: src/libudev/libudev.pc 
> vcpkg/buildtrees/libsystemd/src/v254-832da4078b.clean/tools/meson-render-jinja2.py config.h version.h ../src/v254-832da4078b.clean/src/libudev/libudev.pc.in src/libudev/libudev.pc
> Traceback (most recent call last):
>   File "vcpkg/buildtrees/libsystemd/src/v254-832da4078b.clean/tools/meson-render-jinja2.py", line 9, in <module>
>     import jinja2
> ModuleNotFoundError: No module named 'jinja2'
> [12/191] Generating src/udev/udev.pc with a custom command
> FAILED: src/udev/udev.pc 
> vcpkg/buildtrees/libsystemd/src/v254-832da4078b.clean/tools/meson-render-jinja2.py config.h version.h ../src/v254-832da4078b.clean/src/udev/udev.pc.in src/udev/udev.pc
> Traceback (most recent call last):
>   File "vcpkg/buildtrees/libsystemd/src/v254-832da4078b.clean/tools/meson-render-jinja2.py", line 9, in <module>
>     import jinja2
> ModuleNotFoundError: No module named 'jinja2'
> [13/191] Generating src/basic/arphrd-from-name.gperf with a custom command (wrapped by meson to capture output)
> [14/191] Generating src/basic/af-to-name.h with a custom command (wrapped by meson to capture output)
> [15/191] Generating src/basic/arphrd-to-name.h with a custom command (wrapped by meson to capture output)
> [16/191] Generating src/libsystemd/audit_type-to-name.h with a custom command (wrapped by meson to capture output)
> [17/191] Generating src/basic/cap-from-name.gperf with a custom command (wrapped by meson to capture output)
> [18/191] Generating src/basic/af-from-name.gperf with a custom command (wrapped by meson to capture output)
> [19/191] Generating src/basic/errno-from-name.gperf with a custom command (wrapped by meson to capture output)
> ninja: build stopped: subcommand failed.
>

**Comment by ArthurChan2018** [comment, marked as answer]
@jimwang118
Thanks  a lot
According to [https://github.com/microsoft/vcpkg/issues/35553#issuecomment-1846583132](https://github.com/microsoft/vcpkg/issues/35553#issuecomment-1846583132) I installed python3-jinja2 package and it worked for me

**Reply by Skwangles** [reply]
Yep, can confirm this solved it for me too
